src/functions/module_array.ts

```diff
@@ -269,6 +277,48 @@ export const FREQUENCY: AddFunctionDescription = {
   isExported: true,
 };
 
+// -----------------------------------------------------------------------------
+// SUMPRODUCT
+// -----------------------------------------------------------------------------
+export const SUMPRODUCT: AddFunctionDescription = {
+  description: _lt(
+    "Calculates the sum of the products of corresponding entries in equal-sized ranges."
+  ),
+  args: [
+    arg(
+      "range1 (number, range<number>)",
+      _lt(
+        "The first range whose entries will be multiplied with corresponding entries in the other ranges."
+      )
+    ),
+    arg(
+      "range2 (number, range<number>, repeating)",
+      _lt(
+        "The other range whose entries will be multiplied with corresponding entries in the other ranges."
+      )
+    ),
+  ],
+  returns: ["NUMBER"],
+  compute: function (...args: ArgValue[]): number {
+    assertSameDimensions(_lt("All the ranges must have the same dimensions."), ...args);
+    if (args.every(isMatrix)) {
+      let result = 0;
+      for (let i = 0; i < args[0].length; i++) {
+        for (let j = 0; j < args[0][i].length; j++) {
+          let product = 1;
+          for (const range of args) {
+            product *= toNumber(range[i][j]);
```

Comment:
Let's keep functional consistency:
numbers in string form should not be taken into account when provided by reference

maybe use visitNumbers (may need to change it) to keep consistency